sim/lib/dynamics.py

```diff
@@ -666,19 +670,32 @@ def __push_contact_exposure_events(self, t, infector, base_rate):
         of person `i` (equivalent to `\mu` in model definition)
         """
 
-        def valid_j():
-            '''Generates indices j where `infector` is present
-            at least `self.delta` hours before j '''
-            for j in range(self.n_people):
-                if self.state['susc'][j]:
-                    if self.mob.will_be_in_contact(indiv_i=j, indiv_j=infector, t=t, site=None):
-                        yield j
+        if not self.dynamic_tracing:
+            def valid_j():
+                '''Generates indices j where `infector` is present
+                at least `self.delta` hours before j '''
+                for j in range(self.n_people):
+                    if self.state['susc'][j]:
+                        if self.mob.will_be_in_contact(indiv_i=j, indiv_j=infector, t=t, site=None):
+                            yield j
+
+            valid_contacts = valid_j()
+        else:
+            infectors_contacts = self.mob.find_contacts_of_indiv(self.mob.all_mob_traces, indiv=infector, tmin=t)
```

Comment:
I am not sure of python operates on the the hood, but it would be better note to pass around `self.mob.all_mob_traces` but rather access it within the function itself. Could well be that python handles this by reference but if not (by passing around the object) it would be slower

Reply:
I agree with @larslorch. I think we shouldn't pass self.mob.all_mob_traces but access it within the function since it is cleaner.